compareBuckets/BucketdOplogInterface.js

```diff
@@ -0,0 +1,222 @@
+/*
+ * Main interface for bucketd oplog management
+ *
+ * persist is an interface with the following methods:
+ * - constructor(params)
+ * - load(bucketName, persistData, cb(err, offset))
+ * - save(bucketName, persistData, offset, cb(err))
+
+ * persistData is an interface with the following methods:
+ * - constuctor(params)
+ * - initState(cb(err)): initialize the structure, e.g. initial bucket scan
+ * - loadState(stream, cb(err)): load the state
+ * - saveState(stream, cb(err)): save the state
+ */
+const async = require('async');
+const BucketClient = require('bucketclient').RESTClient;
+const { jsutil } = require('arsenal');
+const LogConsumer = require('arsenal').storage.metadata.bucketclient.LogConsumer;
+const { isMasterKey } = require('arsenal/lib/versioning/Version');
+const Flusher = require('./Flusher');
+const werelogs = require('werelogs');
+
+werelogs.configure({
+    level: 'info',
+    dump: 'error',
+});
+
+class BucketdOplogInterface {
+
+    constructor(params) {
+        let bkBootstrap = ['localhost:9000'];
+        if (params && params.bootstrap) {
+            bkBootstrap = params.bootstrap;
+        }
+        this.bkClient = new BucketClient(bkBootstrap);
+        this.backendRetryTimes = 3;
+        this.backendRetryInterval = 300;
+        this.bucketdOplogQuerySize = 20;
+        this.raftId = 2; // XXX use route
+        this.logger = new werelogs.Logger('BucketdOplogInterface');
+    }
+
+    start(bucketName, persist, persistData, flusherParams, cb) {
```

Comment:
Same as the other function, this is very long and has not that much context or docs.